packages/bitcore-wallet-service/src/lib/chain/xec/index.ts

```diff
@@ -210,6 +215,98 @@ export class XecChain extends BtcChain implements IChain {
     // Get a UTXO
   }
 
+  async burnToken(wallet, mnemonic, tokenId, TOKENQTY, splitTxId) {
+    const tokenInfo = await this.getTokenInfo(tokenId);
+    const rootSeed = await bchjs.Mnemonic.toSeed(mnemonic);
+    // master HDNode
+    let masterHDNode;
+    masterHDNode = bchjs.HDNode.fromSeed(rootSeed);
+    const rootPath = wallet.getRootPath() ? wallet.getRootPath() : "m/44'/1899'/0'";
+    // HDNode of BIP44 account
+    const account = bchjs.HDNode.derivePath(masterHDNode, rootPath);
+    const change = bchjs.HDNode.derivePath(account, '0/0');
+
+    const cashAddress = bchjs.HDNode.toCashAddress(change);
+    const slpAddress = bchjs.HDNode.toSLPAddress(change);
+
+    // Get a UTXO
+    const utxos: UtxoToken[] = await this.getUtxosToken(wallet);
+
+    if (utxos.length === 0) throw new Error('No UTXOs to spend! Exiting.');
+
+    const bchUtxos = _.filter(utxos, item => item.isNonSLP);
+
+    if (bchUtxos.length === 0) {
+      throw new Error('Wallet does not have a BCH UTXO to pay miner fees.');
+    }
+    const tokenUtxos = await this.getTokenUtxos(utxos, tokenInfo);
+
+    if (tokenUtxos.length === 0) {
+      throw new Error('No token UTXOs for the specified token could be found.');
+    }
+
+    const tokenUtxoSelected = tokenUtxos.find(utxo => utxo.txid === splitTxId);
+
+    // Choose a UTXO to pay for the transaction.
+    const bchUtxo = this.findBiggestUtxo(bchUtxos);
+    // console.log(`bchUtxo: ${JSON.stringify(bchUtxo, null, 2)}`);
+
+    // Generate the OP_RETURN code.
+    TOKENQTY = TOKENQTY / Math.pow(10, tokenInfo.decimals);
+    TOKENQTY = _.floor(TOKENQTY, tokenInfo.decimals);
+    const slpData = this.buildBurnOpReturn(tokenInfo.id, new BigNumber(TOKENQTY).times(10 ** tokenInfo.decimals));
+    // BEGIN transaction construction.
+
+    // instance of transaction builder
+    let transactionBuilder;
+    transactionBuilder = new bchjs.TransactionBuilder();
+
+    // Add the BCH UTXO as input to pay for the transaction.
+    const originalAmount = bchUtxo.value;
+    transactionBuilder.addInput(bchUtxo.txid, bchUtxo.outIdx);
```

Comment:
What if the only 1 input is not enough to pay for the transaction?

Reply:
we used biggest utxo amount , so dont need to worry about this case